### PR TITLE
rpc: check dust outputs on account withdrawals and closures

### PR DESCRIPTION
**Pull request by wpaulino**
Prevents https://github.com/lightninglabs/pool/issues/146 from happening.

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -991,10 +992,14 @@ func (s *rpcServer) parseRPCOutputs(outputs []*poolrpc.Output) ([]*wire.TxOut,
 		if err != nil {
 			return nil, err
 		}
-		res = append(res, &wire.TxOut{
+		txOut := &wire.TxOut{
 			Value:    int64(output.ValueSat),
 			PkScript: outputScript,
-		})
+		}
+		if txrules.IsDustOutput(txOut, txrules.DefaultRelayFeePerKb) {
```
Comment (Roasbeef) [inline rpcserver.go:999]: Perhaps we should move this check further into the pipeline (within the account manager) so we can bolster things with some additional unit tests?

**Review by Roasbeef — CHANGES_REQUESTED** [review]

**Review by Roasbeef — APPROVED** [review]
LGTM 🥮

File: account/manager_test.go [hunk 2]
```diff
@@ -857,14 +857,27 @@ func TestAccountWithdrawal(t *testing.T) {
...
 		maxAccountValue, bestHeight+maxAccountExpiry, bestHeight,
 	)
 
-	// With our account created, we'll start our withdrawal by creating the
-	// outputs we'll withdraw our funds to. We'll create three outputs, one
-	// of each supported output type. Each output will have 1/4 of the
-	// account's value.
+	// With our account created, we'll start with an invalid withdrawal to a
+	// dust output, which should fail.
+	dustOutput := &wire.TxOut{Value: 0, PkScript: p2wsh}
```
Comment (Roasbeef) [inline account/manager_test.go:868]: 👍

**Review by halseth — APPROVED** [review]
LGTM 👍

**Pull request merged**